dali/operators/decoder/audio/audio_decoder_test.cc

```diff
@@ -79,14 +80,37 @@ TEST(AudioDecoderTest, WavDecoderTest) {
            << "` to exist";
   }
 
-  auto meta = decoder.Open(make_cspan(bytes));
-  std::vector<DataType> output(meta.length * meta.channels);
-  decoder.DecodeTyped(make_span(output));
+  {
+    auto meta = decoder->Open(make_cspan(bytes));
+    EXPECT_EQ(meta.channels, expected_nchannels);
+    EXPECT_EQ(meta.channels_interleaved, true);
+    EXPECT_EQ(meta.sample_rate, expected_frequency);
+    decoder->Close();
+  }
+
+  {
+    auto meta = decoder->Open(make_cspan(bytes));
+    std::vector<DataType> output(meta.length * meta.channels);
+    decoder->Decode(make_span(output));
+    EXPECT_PRED3(CheckBuffers<DataType>, output.data(), vec.data(), vec.size());
+  }
 
-  EXPECT_EQ(meta.channels, expected_nchannels);
-  EXPECT_EQ(meta.channels_interleaved, true);
-  EXPECT_EQ(meta.sample_rate, expected_frequency);
-  EXPECT_PRED3(CheckBuffers<DataType>, output.data(), vec.data(), vec.size());
+  {
+    auto meta = decoder->Open(make_cspan(bytes));
+    std::vector<DataType> output(meta.length * meta.channels);
+    decoder->DecodeFrames(output.data(), meta.length);
+    EXPECT_PRED3(CheckBuffers<DataType>, output.data(), vec.data(), vec.size());
+  }
+
+  {
+    auto meta = decoder->Open(make_cspan(bytes));
+    int64_t offset = meta.length / 2;
+    int64_t length = meta.length - offset;
+    std::vector<DataType> output(length * meta.channels);
```

Comment:
Maybe you can use larger `output` buffer, fill the end with the dummy data, and see if the DecodeFrames doesn't over run it.